Error in cmake after libtuv update

I try to compile using my de10 nano and get this error .

root@lampa:~/dcurl# make BUILD_FPGA_ACCEL=1
git submodule update --init deps/libtuv
Submodule 'deps/libtuv' (https://github.com/DLTcollab/libtuv.git) registered for path 'deps/libtuv'
Cloning into '/root/dcurl/deps/libtuv'...
Submodule path 'deps/libtuv': checked out '3177b57937056ab5255adc4ad338d5e4ee844eab'
  CC    build/curl.o
  CC    build/constants.o
  CC    build/trinary.o
  CC    build/dcurl.o
  CC    build/implcontext.o
  CC    build/common.o
  CC    build/pow_c.o
  CC    build/pow_fpga_accel.o
make -C deps/libtuv TUV_BUILD_TYPE=release TUV_CREATE_PIC_LIB=yes
make[1]: Entering directory '/root/dcurl/deps/libtuv'
mkdir -p ./build/armv7l-linux/release/cmake
cmake -B./build/armv7l-linux/release/cmake -H./ -DCMAKE_TOOLCHAIN_FILE=./cmake/config/config_armv7l-linux.cmake -DCMAKE_BUILD_TYPE=release -DTARGET_PLATFORM=armv7l-linux -DBUILDTESTER=yes -DBUILD_HOST_HELPER=no -DCREATE_SHARED_LIB=no -DCREATE_PIC_LIB=yes -DTUV_FEATURE_PROCESS=ON
CMake Error at /usr/share/cmake-3.7/Modules/CMakeDetermineSystem.cmake:94 (message):
  Could not find toolchain file: ./cmake/config/config_armv7l-linux.cmake
Call Stack (most recent call first):
  CMakeLists.txt:16 (project)


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
Makefile:63: recipe for target 'all' failed
make[1]: *** [all] Error 1
make[1]: Leaving directory '/root/dcurl/deps/libtuv'
mk/submodule.mk:23: recipe for target 'deps/libtuv/build/armv7l-linux/release/lib/libtuv.o' failed
make: *** [deps/libtuv/build/armv7l-linux/release/lib/libtuv.o] Error 2

On ARM64 machines, I encountered similar problem:
```
$ make
make -C deps/libtuv TUV_BUILD_TYPE=release TUV_CREATE_PIC_LIB=yes
make[1]: Entering directory '/home/ubuntu/dcurl/deps/libtuv'
mkdir -p ./build/aarch64-linux/release/cmake
cmake -B./build/aarch64-linux/release/cmake -H./ -DCMAKE_TOOLCHAIN_FILE=./cmake/config/config_aarch64-linux.cmake -DCMAKE_BUILD_TYPE=release -DTARGET_PLATFORM=aarch64-linux -DBUILDTESTER=yes -DBUILD_HOST_HELPER=no -DCREATE_SHARED_LIB=no -DCREATE_PIC_LIB=yes -DTUV_FEATURE_PROCESS=ON
CMake Error at /usr/share/cmake-3.10/Modules/CMakeDetermineSystem.cmake:100 (message):
  Could not find toolchain file: ./cmake/config/config_aarch64-linux.cmake
Call Stack (most recent call first):
  CMakeLists.txt:16 (project)
```

--- a/Makefile
+++ b/Makefile
@@ -30,16 +30,29 @@ include mk/cpu-features.mk
 # Handle git submodule
 include mk/submodule.mk
 
+# Assign the hardware to CPU if no hardware is specified
+PLATFORMS := $(BUILD_AVX) $(BUILD_SSE) $(BUILD_GENERIC) $(BUILD_GPU) $(BUILD_FPGA_ACCEL)
+ENABLE_PLATFORMS := $(findstring 1,$(PLATFORMS))
+ifneq ("$(ENABLE_PLATFORMS)","1")
+    ifeq ("$(call cpu_feature,AVX)","1")
+        BUILD_AVX = 1
+    else ifeq ("$(call cpu_feature,SSE)","1")
+        BUILD_SSE = 1
+    else
+        BUILD_GENERIC = 1
+    endif
+endif
+
+
 ifeq ("$(BUILD_AVX)","1")
 CFLAGS += -mavx -DENABLE_AVX
 ifeq ("$(call cpu_feature,AVX2)","1")
 CFLAGS += -mavx2
 endif
-else
-BUILD_SSE := $(call cpu_feature,SSE)
-ifeq ("$(BUILD_SSE)","1")
+else ifeq ("$(BUILD_SSE)","1")
 CFLAGS += -msse2 -DENABLE_SSE
-endif
+else ifeq ("$(BUILD_GENERIC)","1")
+CFLAGS += -DENABLE_GENERIC
 endif
 
 ifeq ("$(BUILD_GPU)","1")
@@ -87,12 +100,10 @@ OBJS = \
 
 ifeq ("$(BUILD_AVX)","1")
 OBJS += pow_avx.o
-else
-ifeq ("$(BUILD_SSE)","1")
+else ifeq ("$(BUILD_SSE)","1")
 OBJS += pow_sse.o
-else
+else ifeq ("$(BUILD_GENERIC)","1")
 OBJS += pow_c.o
-endif
 endif
 
 ifeq ("$(BUILD_GPU)","1")
@@ -118,6 +129,14 @@ endif
 
 OBJS := $(addprefix $(OUT)/, $(OBJS))
 
+# Add the libtuv PIC(position independent code) library into the object files
+# if the specified hardware is CPU
+CPU_PLATFORMS := $(BUILD_AVX) $(BUILD_SSE) $(BUILD_GENERIC)
+ENABLE_CPU_PLATFORMS := $(findstring 1,$(CPU_PLATFORMS))
+ifeq ("$(ENABLE_CPU_PLATFORMS)","1")
+    OBJS += $(LIBTUV_LIBRARY)
+endif
+
 $(OUT)/test-%.o: tests/test-%.c $(LIBTUV_PATH)/include
 	$(VECHO) "  CC\t$@\n"
 	$(Q)$(CC) -o $@ $(CFLAGS) -I $(SRC) $(LIBTUV_INCLUDE) -c -MMD -MF $@.d $<
@@ -126,11 +145,11 @@ $(OUT)/%.o: $(SRC)/%.c $(LIBTUV_PATH)/include
 	$(VECHO) "  CC\t$@\n"
 	$(Q)$(CC) -o $@ $(CFLAGS) $(LIBTUV_INCLUDE) -c -MMD -MF $@.d $<
 
-$(OUT)/test-%: $(OUT)/test-%.o $(OBJS) $(LIBTUV_LIBRARY)
+$(OUT)/test-%: $(OUT)/test-%.o $(OBJS)
 	$(VECHO) "  LD\t$@\n"
 	$(Q)$(CC) -o $@ $^ $(LDFLAGS)
 
-$(OUT)/libdcurl.so: $(OBJS) $(LIBTUV_LIBRARY)
+$(OUT)/libdcurl.so: $(OBJS)
 	$(VECHO) "  LD\t$@\n"
 	$(Q)$(CC) -shared -o $@ $^ $(LDFLAGS)
 

--- a/docs/build-n-test.md
+++ b/docs/build-n-test.md
@@ -13,18 +13,23 @@
 ## Build Instructions
 * dcurl allows various combinations of build configurations to fit final use scenarios.
 * You can execute `make config` and then edit file `build/local.mk` for custom build options.
-    - ``BUILD_AVX``: build Intel AVX-accelerated Curl backend.
-    - ``BUILD_GPU``: build OpenCL-based GPU accelerations.
+    - ``BUILD_AVX``: build the Intel AVX-accelerated Curl backend.
+    - ``BUILD_SSE``: build the Intel SSE-accelerated Curl backend.
+    - ``BUILD_GPU``: build the OpenCL-based GPU accelerations.
+    - ``BUILD_FPGA_ACCEL``: build the interface interacting with the Cyclone V FPGA based accelerator. Verified on DE10-nano board and Arrow SoCKit board.
     - ``BUILD_JNI``: build a shared library for IRI. The build system would generate JNI header file
                      downloading from [latest JAVA source](https://github.com/DLTcollab/iri).
     - ``BUILD_COMPAT``: build extra cCurl compatible interface.
-    - ``BUILD_FPGA_ACCEL``: build the interface interacting with the Cyclone V FPGA based accelerator. Verified on DE10-nano board and Arrow SoCKit board.
     - ``BUILD_STAT``: show the statistics of the PoW information.
     - ``BUILD_DEBUG``: dump verbose messages internally.
 * Alternatively, you can specify conditional build as following:
 ```shell
 $ make BUILD_GPU=0 BUILD_JNI=1 BUILD_AVX=1
 ```
+
+## Fallback Build Instructions
+* These instructions do not appear in the file `build/local.mk`.
+    - ``BUILD_GENERIC``: build the generic CPU Curl backend without acceleration.
 
 ## Testing and Validation
 * Test with GPU

--- a/mk/defs.mk
+++ b/mk/defs.mk
@@ -4,17 +4,20 @@ BUILD_DEBUG ?= 0
 # Build AVX backend or not
 BUILD_AVX ?= 0
 
+# Build SSE backend or not
+BUILD_SSE ?= 0
+
 # Build OpenCL backend or not
 BUILD_GPU ?= 0
+
+# Build FPGA backend or not
+BUILD_FPGA_ACCEL ?= 0
 
 # Build JNI glue as the bridge between dcurl and IRI
 BUILD_JNI ?= 0
 
 # Build cCurl compatible interface
 BUILD_COMPAT ?= 0
-
-# Build FPGA backend or not
-BUILD_FPGA_ACCEL ?= 0
 
 # Show the PoW-related statistic messages or not
 BUILD_STAT ?= 0

--- a/src/dcurl.c
+++ b/src/dcurl.c
@@ -21,7 +21,7 @@
 #include "pow_avx.h"
 #elif defined(ENABLE_SSE)
 #include "pow_sse.h"
-#else
+#elif defined(ENABLE_GENERIC)
 #include "pow_c.h"
 #endif
 
@@ -46,7 +46,7 @@ LIST_HEAD(IMPL_LIST);
 extern ImplContext PoWAVX_Context;
 #elif defined(ENABLE_SSE)
 extern ImplContext PoWSSE_Context;
-#else
+#elif defined(ENABLE_GENERIC)
 extern ImplContext PoWC_Context;
 #endif
 
@@ -66,7 +66,7 @@ bool dcurl_init()
     ret &= registerImplContext(&PoWAVX_Context);
 #elif defined(ENABLE_SSE)
     ret &= registerImplContext(&PoWSSE_Context);
-#else
+#elif defined(ENABLE_GENERIC)
     ret &= registerImplContext(&PoWC_Context);
 #endif
 

--- a/tests/common.h
+++ b/tests/common.h
@@ -9,14 +9,15 @@
 #include "pow_avx.h"
 #elif defined(ENABLE_SSE)
 #include "pow_sse.h"
-#elif defined(ENABLE_FPGA_ACCEL)
-#include "pow_fpga_accel.h"
-#else
+#elif defined(ENABLE_GENERIC)
 #include "pow_c.h"
 #endif
 
 #if defined(ENABLE_OPENCL)
 #include "pow_cl.h"
+#endif
+#if defined(ENABLE_FPGA_ACCEL)
+#include "pow_fpga_accel.h"
 #endif
 
 #include <stdint.h>

--- a/tests/test-pow.c
+++ b/tests/test-pow.c
@@ -9,7 +9,7 @@
 extern ImplContext PoWAVX_Context;
 #elif defined(ENABLE_SSE)
 extern ImplContext PoWSSE_Context;
-#else
+#elif defined(ENABLE_GENERIC)
 extern ImplContext PoWC_Context;
 #endif
 
@@ -26,7 +26,7 @@ const char *description[] = {
     "CPU - AVX",
 #elif defined(ENABLE_SSE)
     "CPU - SSE",
-#else
+#elif defined(ENABLE_GENERIC)
     "CPU - pure C",
 #endif
 
@@ -114,7 +114,7 @@ int main()
         PoWAVX_Context,
 #elif defined(ENABLE_SSE)
         PoWSSE_Context,
-#else
+#elif defined(ENABLE_GENERIC)
         PoWC_Context,
 #endif
 

